apollo-router-core/src/json_ext.rs

```diff
@@ -138,9 +47,20 @@ impl ValueExt for Value {
                 }
             }
             (Value::Array(a), Value::Array(mut b)) => {
-                for ((_index, b_value), a_value) in b.drain(..).enumerate().zip(a.iter_mut()) {
+                for (b_value, a_value) in b.drain(..min(a.len(), b.len())).zip(a.iter_mut()) {
```

Comment:
:screwdriver: Zip already zips to the shortest https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.zip

```suggestion
                for (b_value, a_value) in b.drain(..).zip(a.iter_mut()) {
```

Reply:
:+1:

Reply:
I don't know why I assume that I must pass two iterators of the same amount to zip. There must be some old scars of other languages there :sweat_smile:

Reply:
One that fills the missing elements with `null`? :joy: 

There is a zip_longest in itertools if you ever need: https://docs.rs/itertools/0.10.3/itertools/trait.Itertools.html#method.zip_longest

Reply:
ok, so, this does not work, because [drain](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.drain) will drop all the elements that were not consumed once the iterator is dropped, while we actually want to append them afterwards, so I'll revert that change. Do you see another way to fix it?

> One that fills the missing elements with null? 

I'm pretty sure I picked up the habit to check lengths when I worked in C :sweat_smile:

Reply:
:thinking: wow that was tricky!

Ok so I'm all in favor to move to itertool's zip_longest then. The algorithm will be much more readable!

Reply:
I'll take a look at zip_longest, I like the approach of returning an enum